pkg/imgutil/dockerconfigresolver/dockerconfigresolver.go

```diff
@@ -201,7 +201,7 @@ func NewAuthCreds(refHostname string) (AuthCreds, error) {
 			// - ac.ServerAddress:         "https://index.docker.io/v1/".
 			if !isAuthConfigEmpty(ac) {
 				if authConfigHostname == registry.IndexServer {
-					if ac.ServerAddress != registry.IndexServer {
+					if ac.ServerAddress != "" && ac.ServerAddress != registry.IndexServer {
```

Comment:
I think we should enforce the default registry `ac.ServerAddress = registry.IndexServer`  when it is empty for all `refHostname`